pkg/utils/release/components.go

```diff
@@ -0,0 +1,47 @@
+package release
+
+import (
+	"context"
+
+	appstudioApi "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+// Contains all methods related with component objects CRUD operations.
+type ComponentsInterface interface {
+	// Creates a component based on container image.
+	CreateComponentWithDockerSource(applicationName, componentName, namespace, gitSourceURL, containerImageSource, outputContainerImage, secret string) (*appstudioApi.Component, error)
+}
+
+// CreateComponentWithDockerSource creates a component based on container image source.
+func (r *releaseFactory) CreateComponentWithDockerSource(applicationName, componentName, namespace, gitSourceURL, containerImageSource, outputContainerImage, secret string) (*appstudioApi.Component, error) {
```

Comment:
This should not be here. Should be in utils/has/components.go

Reply:
Moved.